infrastructure/iac/iac.go

```diff
@@ -117,12 +153,16 @@ func (iac *Scanner) doScan(ctx context.Context, documentURI sglsp.DocumentURI, w
 	s := iac.instrumentor.StartSpan(ctx, method)
 	defer iac.instrumentor.Finish(s)
 
-	iac.mutex.Lock()
-	defer iac.mutex.Unlock()
+	// iac.mutex.Lock()
```

Comment:
This can be removed instead of commented.

Reply:
I'll uncomment these because of the IAC restriction (+ added comment)